libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/encryption/RustEncryptionService.kt

```diff
@@ -66,10 +65,15 @@ internal class RustEncryptionService(
     private val enableRecoveryProgressMapper = EnableRecoveryProgressMapper()
     private val backupUploadStateMapper = BackupUploadStateMapper()
     private val steadyStateExceptionMapper = SteadyStateExceptionMapper()
-    private var backupStateListenerTaskHandle: TaskHandle? = null
-    private var recoveryStateListenerTaskHandle: TaskHandle? = null
 
-    private val backupStateFlow = MutableStateFlow(service.backupState().let(backupStateMapper::map))
+    private val backupStateFlow = mxCallbackFlow {
+        val listener = object : BackupStateListener {
+            override fun onUpdate(status: RustBackupState) {
+                trySend(backupStateMapper.map(status))
+            }
+        }
+        service.backupStateListener(listener)
```

Comment:
I'd have offer this as extension methods as we already do that in several places (RoomList, Timeline...)

Reply:
Yes, sorry, I forgot this: 7d4cf6d574c7ed51a72dff11d1fa875bc34d0549